components/infra-proxy-service/migrations/pipeline/utility.go

```diff
@@ -305,39 +308,169 @@ func Unzip(ctx context.Context, mst storage.MigrationStorage, result pipeline.Re
 			file.Mode())
 		if err != nil {
 			log.Errorf("cannot create a file: %s.", err)
-			mst.FailedUnzip(ctx, result.Meta.MigrationID, result.Meta.ServerID, "cannot create a file", 0, 0, 0)
+			_, _ = mst.FailedUnzip(ctx, result.Meta.MigrationID, result.Meta.ServerID, "cannot create a file", 0, 0, 0)
```

Comment:
As discussed,instead of adding the database entries in utility functions we are adding it in the pipeline